Build image failed w/o a clear error message with Maven plugin v0.9.7

<!-- Thanks for filing an issue! Please use this template to provide more context for the issue. -->

**Description of the issue**: Build image failed w/o a clear error message with Maven plugin v0.9.7

**Expected behaviour**: An image is built and pushed to Docker hub.

**Steps to reproduce**: 
* Set up docker hub credentials in Maven's `settings.xml`
* Run `mvn compile jib:build` in the project root

**Environment**: 
* Windows 10
* Maven 3.5.3
* Oracle JDK 1.8.0_172

**`jib-maven-plugin` Configuration**: 
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>io.plagov</groupId>
    <artifactId>jib-docker-demo</artifactId>
    <version>1.0-SNAPSHOT</version>
    <build>
        <plugins>
            <plugin>
                <groupId>com.google.cloud.tools</groupId>
                <artifactId>jib-maven-plugin</artifactId>
                <version>0.9.7</version>
                <configuration>
                    <to>
                        <image>registry.hub.docker.com/vplagov/image-built-with-jib</image>
                    </to>
                </configuration>
            </plugin>
        </plugins>
    </build>
</project>
```
Configuration in `settings.xml`:
```xml
<settings>
...
	<servers>
		<server>
			<id>registry.hub.docker.com</id>
			<username>my_username</username>
			<password>my_password</password>
		</server>
	</servers>
...
</settings>
```

**Log output**: full output after I run `mvn compile jib:build` command:
```
[INFO] Scanning for projects...
[INFO]
[INFO] ---------------------< io.plagov:jib-docker-demo >----------------------
[INFO] Building jib-docker-demo 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ jib-docker-demo ---
[WARNING] Using platform encoding (Cp1257 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] Copying 0 resource
[INFO]
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ jib-docker-demo ---
[INFO] Changes detected - recompiling the module!
[WARNING] File encoding has not been set, using platform encoding Cp1257, i.e. build is platform dependent!
[INFO] Compiling 1 source file to C:\Users\vitali.plagov\IdeaProjects\jib-docker-demo\target\classes
[INFO]
[INFO] --- jib-maven-plugin:0.9.7:build (default-cli) @ jib-docker-demo ---
[WARNING] Base image 'gcr.io/distroless/java' does not use a specific image digest - build may not be reproducible
[INFO]
[INFO] Containerizing application to vplagov/image-built-with-jib...
[INFO]
[INFO] Retrieving registry credentials for registry.hub.docker.com...
[INFO] Getting base image gcr.io/distroless/java...
[INFO] Building dependencies layer...
[INFO] Building resources layer...
[INFO] Building classes layer...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 3.735 s
[INFO] Finished at: 2018-07-26T09:27:02+03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:0.9.7:build (default-cli) on project jib-docker-demo: Build image failed: C:\Users\vitali.plagov\IdeaProjects\jib-docker-demo\target\jib-cache\5063664208194952830.tmp -> C:\Users\vitali.plagov\IdeaProjects\jib-docker-demo\target\jib-cache\252a131556a09f39f95d58172fca18c51b4ba688a9c29b0fc1bdd118f8e9abb2.tar.gz -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

**Additional Information**: I tried with several other versions, like 0.9.0, 0.9.4 and 0.9.7. Unfortunatelly, still more or less the same output. Also, the day before I've managed to build an image and push it to Docker hub. I got an error with an authentication when I encrypted the password with Maven, but after I put my DockerID password in a plain text, then I've succesfully built and pushed an image.

Hi @chanseokoh,
Here's the output of `mvn -e clean compile jib:build`:

```
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO]
[INFO] ---------------------< io.plagov:jib-docker-demo >----------------------
[INFO] Building jib-docker-demo 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ jib-docker-demo ---
[INFO] Deleting C:\Users\vitali.plagov\IdeaProjects\jib-docker-demo\target
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ jib-docker-demo ---
[WARNING] Using platform encoding (Cp1257 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] Copying 0 resource
[INFO]
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ jib-docker-demo ---
[INFO] Changes detected - recompiling the module!
[WARNING] File encoding has not been set, using platform encoding Cp1257, i.e. build is platform dependent!
[INFO] Compiling 1 source file to C:\Users\vitali.plagov\IdeaProjects\jib-docker-demo\target\classes
[INFO]
[INFO] --- jib-maven-plugin:0.9.7:build (default-cli) @ jib-docker-demo ---
[WARNING] Base image 'gcr.io/distroless/java' does not use a specific image digest - build may not be reproducible
[INFO]
[INFO] Containerizing application to vplagov/image-built-with-jib...
[INFO]
[INFO] Retrieving registry credentials for registry.hub.docker.com...
[INFO] Getting base image gcr.io/distroless/java...
[INFO] Building dependencies layer...
[INFO] Building classes layer...
[INFO] Building resources layer...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 4.078 s
[INFO] Finished at: 2018-07-26T15:09:57+03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:0.9.7:build (default-cli) on project jib-docker-demo: Build image failed: C:\Users\vitali.plagov\IdeaProjects\jib-docker-demo\target\jib-cache\660889960859161559.tmp -> C:\Users\vitali.plagov\IdeaProjects\jib-docker-demo\target\jib-cache\252a131556a09f39f95d58172fca18c51b4ba688a9c29b0fc1bdd118f8e9abb2.tar.gz -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.google.cloud.tools:jib-maven-plugin:0.9.7:build (default-cli) on project jib-docker-demo: Build image failed
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Build image failed
    at com.google.cloud.tools.jib.maven.BuildImageMojo.execute (BuildImageMojo.java:165)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.nio.file.FileAlreadyExistsException: C:\Users\vitali.plagov\IdeaProjects\jib-docker-demo\target\jib-cache\660889960859161559.tmp -> C:\Users\vitali.plagov\IdeaProjects\jib-docker-demo\target\jib-cache\252a131556a09f39f95d58172fca18c51b4ba688a9c29b0fc1bdd118f8e9abb2.tar.gz
    at sun.nio.fs.WindowsException.translateToIOException (WindowsException.java:81)
    at sun.nio.fs.WindowsException.rethrowAsIOException (WindowsException.java:97)
    at sun.nio.fs.WindowsFileCopy.move (WindowsFileCopy.java:387)
    at sun.nio.fs.WindowsFileSystemProvider.move (WindowsFileSystemProvider.java:287)
    at java.nio.file.Files.move (Files.java:1395)
    at com.google.cloud.tools.jib.cache.CacheWriter.writeLayer (CacheWriter.java:83)
    at com.google.cloud.tools.jib.builder.steps.BuildAndCacheApplicationLayerStep.call (BuildAndCacheApplicationLayerStep.java:165)
    at com.google.cloud.tools.jib.builder.steps.BuildAndCacheApplicationLayerStep.call (BuildAndCacheApplicationLayerStep.java:38)
    at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly (TrustedListenableFutureTask.java:127)
    at com.google.common.util.concurrent.InterruptibleTask.run (InterruptibleTask.java:57)
    at com.google.common.util.concurrent.TrustedListenableFutureTask.run (TrustedListenableFutureTask.java:80)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:624)
    at java.lang.Thread.run (Thread.java:748)
[ERROR]
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

It throws a `FileAlreadyExistsException` but I can't understand how can it be if a `clean` goal cleans all previous build files. I even run `mvn clean` separately, checked if the workspace is indeed clean and nothing except .java files are present.

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheWriter.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheWriter.java
@@ -80,7 +80,7 @@ public class CacheWriter {
       Path layerFile = getLayerFile(compressedBlobDescriptor.getDigest());
       try {
         Files.move(tempLayerFile, layerFile);
-      } catch (FileAlreadyExistsException ex) {
+      } catch (FileAlreadyExistsException ignored) {
         // If the file already exists, we skip renaming and use the existing file. This happens if a
         // new layer happens to have the same content as a previously-cached layer.
         //

--- a/jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheWriter.java
+++ b/jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheWriter.java
@@ -27,9 +27,9 @@ import java.io.BufferedInputStream;
 import java.io.BufferedOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.nio.file.FileAlreadyExistsException;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.nio.file.StandardCopyOption;
 import java.nio.file.attribute.FileTime;
 import java.time.Instant;
 import java.util.zip.GZIPInputStream;
@@ -76,11 +76,17 @@ public class CacheWriter {
       compressorStream.close();
       BlobDescriptor compressedBlobDescriptor = compressedDigestOutputStream.toBlobDescriptor();
 
-      // Renames the temporary layer file to the correct filename. If the file already exists, we
-      // skip renaming and use the existing file. This happens if a new layer happens to have the
-      // same content as a previously-cached layer.
+      // Renames the temporary layer file to the correct filename.
       Path layerFile = getLayerFile(compressedBlobDescriptor.getDigest());
-      Files.move(tempLayerFile, layerFile, StandardCopyOption.REPLACE_EXISTING);
+      try {
+        Files.move(tempLayerFile, layerFile);
+      } catch (FileAlreadyExistsException ex) {
+        // If the file already exists, we skip renaming and use the existing file. This happens if a
+        // new layer happens to have the same content as a previously-cached layer.
+        //
+        // Do not attempt to remove the try-catch block with the idea of checking file existence
+        // before moving; there can be concurrent file moves.
+      }
 
       CachedLayer cachedLayer = new CachedLayer(layerFile, compressedBlobDescriptor, diffId);
       LayerMetadata layerMetadata =

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/cache/CacheWriterTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/cache/CacheWriterTest.java
@@ -25,19 +25,18 @@ import com.google.cloud.tools.jib.image.LayerEntry;
 import com.google.cloud.tools.jib.image.ReproducibleLayerBuilder;
 import com.google.cloud.tools.jib.image.UnwrittenLayer;
 import com.google.common.collect.ImmutableList;
-import com.google.common.io.CharStreams;
+import com.google.common.io.ByteStreams;
 import com.google.common.io.CountingOutputStream;
 import com.google.common.io.Resources;
 import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
-import java.io.InputStreamReader;
 import java.net.URISyntaxException;
-import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.Collections;
+import java.util.stream.Stream;
 import java.util.zip.GZIPInputStream;
 import java.util.zip.GZIPOutputStream;
 import org.junit.Assert;
@@ -50,9 +49,14 @@ import org.mockito.Mockito;
 /** Tests for {@link CacheWriter}. */
 public class CacheWriterTest {
 
-  @Rule public TemporaryFolder temporaryCacheDirectory = new TemporaryFolder();
+  private static boolean isTarGz(Path path) {
+    return path.toString().endsWith(".tar.gz");
+  }
+
+  @Rule public final TemporaryFolder temporaryCacheDirectory = new TemporaryFolder();
 
   private Cache testCache;
+  private CacheWriter cacheWriter;
 
   private Path resourceBlob;
 
@@ -74,17 +78,14 @@ public class CacheWriterTest {
     Path cacheDirectory = temporaryCacheDirectory.newFolder().toPath();
 
     testCache = Cache.init(cacheDirectory);
+    // Writes resourceBlob as a layer to the cache.
+    cacheWriter = new CacheWriter(testCache);
 
     resourceBlob = Paths.get(Resources.getResource("blobA").toURI());
   }
 
   @Test
   public void testWriteLayer_unwritten() throws IOException {
-    ExpectedLayer expectedLayer = getExpectedLayer();
-
-    // Writes resourceBlob as a layer to the cache.
-    CacheWriter cacheWriter = new CacheWriter(testCache);
-
     UnwrittenLayer unwrittenLayer = new UnwrittenLayer(Blobs.from(resourceBlob));
 
     ReproducibleLayerBuilder mockReproducibleLayerBuilder =
@@ -94,33 +95,62 @@ public class CacheWriterTest {
         .thenReturn(
             ImmutableList.of(
                 new LayerEntry(
-                    ImmutableList.of(Paths.get("some", "source", "file")),
-                    "/some/extraction/path")));
+                    ImmutableList.of(Paths.get("some/source/file")), "/some/extraction/path")));
 
     CachedLayerWithMetadata cachedLayerWithMetadata =
         cacheWriter.writeLayer(mockReproducibleLayerBuilder);
     testCache.addCachedLayersWithMetadataToMetadata(
         Collections.singletonList(cachedLayerWithMetadata));
 
-    CachedLayerWithMetadata layerInMetadata = testCache.getUpdatedMetadata().getLayers().get(0);
-    Assert.assertNotNull(layerInMetadata.getMetadata());
-    Assert.assertEquals(1, layerInMetadata.getMetadata().getEntries().size());
+    LayerMetadata layerMetadata = testCache.getUpdatedMetadata().getLayers().get(0).getMetadata();
+    Assert.assertNotNull(layerMetadata);
+    Assert.assertEquals(1, layerMetadata.getEntries().size());
     Assert.assertEquals(
-        Collections.singletonList(Paths.get("some", "source", "file").toString()),
-        layerInMetadata.getMetadata().getEntries().get(0).getSourceFilesStrings());
+        Collections.singletonList(Paths.get("some/source/file").toString()),
+        layerMetadata.getEntries().get(0).getSourceFilesStrings());
     Assert.assertEquals(
-        "/some/extraction/path",
-        layerInMetadata.getMetadata().getEntries().get(0).getExtractionPath());
+        "/some/extraction/path", layerMetadata.getEntries().get(0).getExtractionPath());
 
-    verifyCachedLayerIsExpected(expectedLayer, cachedLayerWithMetadata);
+    verifyCachedLayerIsExpected(getExpectedLayer(), cachedLayerWithMetadata);
+  }
+
+  // Windows file overwrite issue: https://github.com/GoogleContainerTools/jib/issues/719
+  @Test
+  public void testWriteLayer_doesNotOverwriteExistingTarGz() throws IOException, InterruptedException {
+    // Writes resourceBlob as a layer to the cache.
+    UnwrittenLayer unwrittenLayer = new UnwrittenLayer(Blobs.from(resourceBlob));
+
+    ReproducibleLayerBuilder layerBuilder = Mockito.mock(ReproducibleLayerBuilder.class);
+    Mockito.when(layerBuilder.build()).thenReturn(unwrittenLayer);
+    LayerEntry layerEntry =
+        new LayerEntry(ImmutableList.of(Paths.get("some/source/file")), "/some/extraction/path");
+    Mockito.when(layerBuilder.getLayerEntries()).thenReturn(ImmutableList.of(layerEntry));
+
+    cacheWriter.writeLayer(layerBuilder);
+    Assert.assertEquals(1, getTarGzCountInCache());
+    long tarGzModifiedTime = getTarGzModifiedTimeInCache();
+
+    Thread.sleep(1000); // to have different modified time
+    cacheWriter.writeLayer(layerBuilder);
+    Assert.assertEquals(1, getTarGzCountInCache());
+    Assert.assertEquals(tarGzModifiedTime, getTarGzModifiedTimeInCache());
+  }
+
+  private long getTarGzCountInCache() throws IOException {
+    try (Stream<Path> stream = Files.walk(temporaryCacheDirectory.getRoot().toPath())) {
+      return stream.filter(CacheWriterTest::isTarGz).count();
+    }
+  }
+
+  private long getTarGzModifiedTimeInCache() throws IOException {
+    try (Stream<Path> stream = Files.walk(temporaryCacheDirectory.getRoot().toPath())) {
+      return stream.filter(CacheWriterTest::isTarGz).findFirst().get().toFile().lastModified();
+    }
   }
 
   @Test
   public void testGetLayerOutputStream() throws IOException {
     ExpectedLayer expectedLayer = getExpectedLayer();
-
-    // Writes resourceBlob as a layer to the cache.
-    CacheWriter cacheWriter = new CacheWriter(testCache);
 
     CountingOutputStream layerOutputStream =
         cacheWriter.getLayerOutputStream(expectedLayer.blobDescriptor.getDigest());
@@ -142,9 +172,6 @@ public class CacheWriterTest {
    *     file
    */
   private ExpectedLayer getExpectedLayer() throws IOException {
-    String expectedBlobAString =
-        new String(Files.readAllBytes(resourceBlob), StandardCharsets.UTF_8);
-
     // Gets the expected content descriptor, diff ID, and compressed BLOB.
     ByteArrayOutputStream compressedBlobOutputStream = new ByteArrayOutputStream();
     CountingDigestOutputStream compressedDigestOutputStream =
@@ -152,7 +179,8 @@ public class CacheWriterTest {
     CountingDigestOutputStream uncompressedDigestOutputStream;
     try (GZIPOutputStream compressorStream = new GZIPOutputStream(compressedDigestOutputStream)) {
       uncompressedDigestOutputStream = new CountingDigestOutputStream(compressorStream);
-      uncompressedDigestOutputStream.write(expectedBlobAString.getBytes(StandardCharsets.UTF_8));
+      byte[] expectedBlobABytes = Files.readAllBytes(resourceBlob);
+      uncompressedDigestOutputStream.write(expectedBlobABytes);
     }
 
     BlobDescriptor expectedBlobADescriptor = compressedDigestOutputStream.toBlobDescriptor();
@@ -171,15 +199,10 @@ public class CacheWriterTest {
     // Reads the cached layer back.
     Path compressedBlobFile = cachedLayer.getContentFile();
 
-    try (InputStreamReader fileReader =
-        new InputStreamReader(
-            new GZIPInputStream(Files.newInputStream(compressedBlobFile)),
-            StandardCharsets.UTF_8)) {
-      String decompressedString = CharStreams.toString(fileReader);
-
-      String expectedBlobAString =
-          new String(Files.readAllBytes(resourceBlob), StandardCharsets.UTF_8);
-      Assert.assertEquals(expectedBlobAString, decompressedString);
+    try (GZIPInputStream in = new GZIPInputStream(Files.newInputStream(compressedBlobFile))) {
+      byte[] decompressedBytes = ByteStreams.toByteArray(in);
+      byte[] expectedBlobABytes = Files.readAllBytes(resourceBlob);
+      Assert.assertArrayEquals(expectedBlobABytes, decompressedBytes);
       Assert.assertEquals(
           expectedLayer.blobDescriptor.getSize(), cachedLayer.getBlobDescriptor().getSize());
       Assert.assertEquals(

--- a/jib-core/src/test/java/com/google/cloud/tools/jib/cache/CacheWriterTest.java
+++ b/jib-core/src/test/java/com/google/cloud/tools/jib/cache/CacheWriterTest.java
@@ -116,7 +116,8 @@ public class CacheWriterTest {
 
   // Windows file overwrite issue: https://github.com/GoogleContainerTools/jib/issues/719
   @Test
-  public void testWriteLayer_doesNotOverwriteExistingTarGz() throws IOException, InterruptedException {
+  public void testWriteLayer_doesNotOverwriteExistingTarGz()
+      throws IOException, InterruptedException {
     // Writes resourceBlob as a layer to the cache.
     UnwrittenLayer unwrittenLayer = new UnwrittenLayer(Blobs.from(resourceBlob));
 

